networkx/algorithms/simple_paths.py

```diff
@@ -68,17 +68,24 @@ def is_simple_path(G, nodes):
     False
 
     """
+    assert isinstance(nodes, list), "Object passed as `nodes` must be a list."
```

Comment:
-1 on this assertion - the docstring is clear on the expected type, and assertions are really intended to check that scenarios that should not ever exist are not hit.

Reply:
Thanks, makes sense!